releases/release-5.1.4.md

```diff
@@ -0,0 +1,165 @@
+        - Try to fix owner stuck caused by etcd txn timeout or etcd watch channel blocked in EtcdWorker. [#3615](https://github.com/pingcap/tiflow/issues/3615)
+        - Fix kvclient takes too long time to recover [#3191](https://github.com/pingcap/tiflow/issues/3191)
+        (dup) - Fix the issue that Avro sink does not support parsing JSON type columns [#3624](https://github.com/pingcap/tiflow/issues/3624)
+        (dup) - Fix the negative value error in the changefeed checkpoint lag [#3010](https://github.com/pingcap/ticdc/issues/3010)
+        (dup) - Fix OOM in container environments [#1798](https://github.com/pingcap/ticdc/issues/1798)
+        (dup) - Fix the TiCDC replication interruption issue when multiple TiKVs crash or during a forced restart [#3288](https://github.com/pingcap/ticdc/issues/3288)
+        (dup) - Fix the memory leak issue after processing DDLs [#3174](https://github.com/pingcap/ticdc/issues/3174)
+        (dup) - Fix the issue that changefeed does not fail fast enough when the ErrGCTTLExceeded error occurs [#3111](https://github.com/pingcap/ticdc/issues/3111)
+        (dup) - Optimize rate limiting control on TiKV reloads to reduce gPRC congestion during changefeed initialization [#3110](https://github.com/pingcap/ticdc/issues/3110)
+        (dup) (dup) - Fix the issue that Kafka may send excessively large messages by setting the default value of `max-message-bytes` to `10M` [#3081](https://github.com/pingcap/tiflow/issues/3081)
+        (dup) - Fix the issue that TiCDC replication task might terminate when the upstream TiDB instance unexpectedly exits [#3061](https://github.com/pingcap/tiflow/issues/3061)
+        (dup) - Fix the issue that TiCDC process might panic when TiKV sends duplicate requests to the same Region [#2386](https://github.com/pingcap/tiflow/issues/2386)
+        (dup) - Fix the issue that the volume of Kafka messages generated by TiCDC is not constrained by `max-message-size` [#2962](https://github.com/pingcap/tiflow/issues/2962)
+        - Nond [#2983](https://github.com/pingcap/tiflow/issues/2983)
+        - Add metrics to observe incremental scan remaining time [#2985](https://github.com/pingcap/tiflow/issues/2985)
+        (dup) - Fix the issue that TiCDC sync task might pause when an error occurs during writing a Kafka message [#2978](https://github.com/pingcap/tiflow/issues/2978)
+
+## Improvements
+
++ TiDB
+
+    - planner: support column range partition pruning for builtin function IN [#26739](https://github.com/pingcap/tidb/issues/26739)
+    - Track the memory usage of IndexJoin more accurate. [#28650](https://github.com/pingcap/tidb/issues/28650)
+
++ TiFlash
+
+    - support functions of ADDDATE() and DATE_ADD() pushed down to tiflash
+
+## Bug Fixes
+
++ TiDB
+
+    - Fix the bug that indexHashJoin may return the error `send on closed channel`. [#31129](https://github.com/pingcap/tidb/issues/31129)
```

Comment:
- Fix the memory leak bug when using analyze version 2(@@tidb_analyze_version = 2). [#29305](https://github.com/pingcap/tidb/pull/29305)
    - Fix the issue that `MaxDays` and `MaxBackups` not working for slow log. [#25716](https://github.com/pingcap/tidb/issues/25716)
    - Fix the panic that may happen when using `ON DUPLICATE KEY UPDATE`. [#28078](https://github.com/pingcap/tidb/issues/28078)
    - Fix the wrong result when using join with enum type [#27831](https://github.com/pingcap/tidb/issues/27831)
    - Fix the issue that `IndexHashJoin` may return the error `send on closed channel`. [#31129](https://github.com/pingcap/tidb/issues/31129)
    - Fix the issue that recycle idle connection of ['BatchCommands'](https://docs.pingcap.com/tidb/stable/tidb-configuration-file/#max-batch-size) may block sending requests in some rare cases. [#27678](https://github.com/pingcap/tidb/pull/27678)